### ISSUE TITLE
Setting deprecated type field on match queries

### ISSUE THREAD

**Issue by diogogmt** [olivere/elastic#609]
Please use the following questions as a guideline to help me answer
your issue/question without further inquiry. Thank you.

### Which version of Elastic are you using?

[ ] elastic.v2 (for Elasticsearch 1.x)
[ ] elastic.v3 (for Elasticsearch 2.x)
[x] elastic.v5 (for Elasticsearch 5.x)

### Please describe the expected behavior

When building a `MatchPhrase` or `MatchPhraseQuery` the output JSON should have the objects with keys of `match_phrase` and `match_phrase_query` instead of `match` with a `type`

Eg;
```json
{
  "match_phrase": {
    "title": {
      "boost": 1,
      "query": "test"
    }
  }
}
```

Instead of 
```json
{
  "match": {
    "title": {
      "boost": 1,
      "query": "test",
      "type": "phrase"
    }
  }
},
```

https://www.elastic.co/guide/en/elasticsearch/reference/5.6/query-dsl-match-query-phrase.html

### Please describe the actual behavior

Generated JSON query contains deprecated key.
The following warning is displayed in the logs: `[WARN ][o.e.d.c.ParseField       ] Deprecated field [type] used, replaced by [match_phrase and match_phrase_prefix query]`

### Any steps to reproduce the behavior?

Build a `MatchPhrase` or `MatchPhrasePrefix` query.
Eg; `elastic.NewMatchPhraseQuery("key", "val")`

**Comment by olivere** [comment]
Thanks. I will address this in the next update. #607 seems to be related (or even the same issue).

**Comment by olivere** [comment]
I fixed this in 4ca1c1181fb40a6ca2cab98f88e08247e2d6ba52 or version 5.0.47.
